docs/polymorfisme.md

```diff
@@ -0,0 +1,299 @@
+          type: integer
+          minimum: 1
+          maximum: 12
+  required:
+    - jaar
+    - maand
+JaarDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        jaar:
+          type: integer
+          maximum: 9999
+  required:
+    - jaar
+OnbekendDatum:
+  allOf:
+    - $ref: '#/components/schemas/AbstractDatum'
+    - type: object
+      properties:
+        indicatieOnbekend:
+          type: bool
+  required:
+    - indicatieOnbekend
+```
+
+In de nieuwe datum definities zijn de type enum waarden verwijderd. Deze zijn niet meer nodig omdat standaard de naam van de afgeleide schemas (in dit geval: VolledigeDatum,JaarMaandDatum,JaarDatum,OnbekendDatum) als enum waarde wordt gebruikt. Van deze standaard 'mapping' kan worden afgeweken door gebruik te maken van het **mapping** element van de discriminator. In bovenstaande definities is gebruik gemaakt van het **mapping** element omdat 'Datum' als type waarde is gekozen voor een VolledigeDatum type.
+
+Het toepassen van polymorfisme in de OpenAPI specificatie van de BRP API leidt tot een duidelijker API contract definitie. Er kan op een eenduidige manier worden afgeleid welke data types er aan een veld kan worden toegekend, wat de velden zijn voor elk data type, welke van de velden verplicht zijn en welke optioneel. Hierdoor kan tooling worden gebruikt om de request en response van de API te valideren, zodat er geen onjuiste data bij de consumers en bij de provider terecht kan komen.
+
+Ook wordt de logica, die nodig is om te bepalen om welk data type het gaat, simpeler. Alleen het **discriminator** veld is nodig om de data type te kunnen bepalen. Wanneer een code generator wordt gebruikt die polymorfisme ondersteunt, dan kan deze logica door de code generator worden gegenereerd en door de code generator aan het serialisatie/deserialisatie proces worden toegevoegd. Dit houdt de business logica vrij van de nodige 'plumbling' code.
+
+Waar wel rekening mee gehouden moet worden is dat niet alle OpenAPI tooling polymorfisme ondersteunt/zal ondersteunen. Zo kan Swagger UI een OpenAPI specificatie waarin polymorfisme is toegepast (nog) niet goed tonen. Ook de Swagger code generator en Postman ondersteunen polymorfisme (nog) niet. Er moet dan worden gekeken naar alternatieven, zoals [ReDoc](https://redocly.github.io/redoc/), [OpenAPI Generator](https://openapi-generator.tech/) en [NSwag](https://github.com/RicoSuter/NSwag).
+
+Verder brengt het toepassen van polymorfisme een zekere mate van complexiteit met zich mee. Voor een simpele type als de datum type met slechts vier velden kun je je afvragen of het noodzakelijk is om vijf datum types te introduceren in plaats van één datum type. Voor de Verblijfplaats type biedt het zeker meerwaarde omdat elk Verblijfplaats type meerdere velden heeft die niet bij een andere Verblijfplaats type voorkomt. Ook is de logica om te bepalen om welk Verblijfplaats type het gaat veel complexer.
```

Comment:
Moet je hier zelf al twijfels plaatsen bij het gebruik van polymorfisme bij datum types? We hadden daar een goede reden voor, toch? Men zal wellicht redeneren dat het bedenkelijk is om hier polymorfisme te introduceren als wij er zelf al aan twijfelen.